src/Microsoft.AspNetCore.Mvc.Core/Internal/DefaultValidationMetadataProvider.cs

```diff
@@ -34,6 +36,26 @@ public void CreateValidationMetadata(ValidationMetadataProviderContext context)
                     }
                 }
             }
+
+            // [ValidateNever] on a type affects properties in that type, not properties that have that type. Thus,
+            // we ignore context.TypeAttributes for properties and don't check at all for types.
```

Comment:
Yes, this is consistent with `[BindNever]` and friends. I copied the binding provider code from there. And, this is exactly the area where the `[BindNever]` docs were ambiguous.